docs/error.md

```diff
@@ -0,0 +1,52 @@
+# Error
+
+An object representing an error that can be triggered / thrown.
+
+
+## **Overview**
+
+The errors are used to notify the user of a critical issue or log a recoverable one. An error object is consisted of three mandatory parameters: error [severity](severity), error [category](category) and an error [code](code). The forth parameter is an optional object with more information about the issue.
+
+There are two severities to an error:
+critical and recoverable.
+A critical error is triggered in an error event, and is shown to the user as error overlay in the player itself.
+A recoverable error is not shown to an end user, but appears in the console.
+
+The errors are divided into nine categories, like network and DRM issues.
+
+Each category has a number of codes that describe the problem more specifically.
+
+---------
+
+## **Creating an Error object**
+
+
+Creating an error object:
+
+```javascript
+
+new Error(Error.Severity.CRITICAL,
+		    Error.Category.NETWORK,
+		    Error.Code.HTTP_ERROR,
+		     {'url': 'www.some-bad-url.com'})
+```
+
+
+## Registering to an error event
+```javascript
+ player.addEventListener(player.Event.ERROR, e => {
+	 const error = e.payload;
+	 const severity = error.severity;
+	 const category = error.category;
+	 const code = error.code;
+	 const moreInfo = error.data;
+ })
+```
+
+## **Error string**
```

Comment:
I don't think this section and title make much sense.
I understand what you are trying to say but don't think we are saying it well enough.
Need to make it more like - if you want to debug player and see explicit error messages then you need to use debug mode which will enable seeing explicit error messages in console.
This is just general, not actual words I will use.
you can consult with @AnnaAleksandrowicz if you don't have a better way to express this.

Also, I would connect this to the debug article we are adding, as we are mentioning it here, but it's not clear how to set it in this scope. @dan-ziv FYI.